src/RestControllers/SMART/SMARTAuthorizationController.php

```diff
@@ -162,9 +162,10 @@ public function patientSelectConfirm()
 
         // set our patient information up in our pid so we can handle our code property...
         try {
-            $patient_id = $_POST['patient_id'];
+            $patient_id = $_POST['patient_id']; // this patient_id is actually a uuid.. wierd
             $searchController = new PatientContextSearchController(new PatientService(), $this->logger);
-            $searchController->getPatientForUser($patient_id, $user_uuid);
+            // throws access denied if user doesn't have access
+            $foundPatient = $searchController->getPatientForUser($patient_id, $user_uuid);
             // put PID in session
             $_SESSION['pid'] = $patient_id;
```

Comment:
this inconsistency will likely be problematic in backend openemr stuff if the session pid is set to the puuid. For things to work (backend code), this should be the pid and not the puuid.

Reply:
Probably should store both if session is referenced for the uuid at some point in transaction. Not sure why that'd be the case though....